absl/random/internal/randen_detect.cc

```diff
@@ -31,14 +31,15 @@
 
 #if defined(ABSL_ARCH_X86_64)
 #define ABSL_INTERNAL_USE_X86_CPUID
-#elif defined(ABSL_ARCH_PPC) || defined(ABSL_ARCH_ARM) || \
-    defined(ABSL_ARCH_AARCH64)
+#elif defined(ABSL_ARCH_PPC) || defined(ABSL_ARCH_ARM) || defined(ABSL_ARCH_AARCH64)
 #if defined(__ANDROID__)
 #define ABSL_INTERNAL_USE_ANDROID_GETAUXVAL
 #define ABSL_INTERNAL_USE_GETAUXVAL
 #elif defined(__linux__) && defined(ABSL_HAVE_GETAUXVAL)
 #define ABSL_INTERNAL_USE_LINUX_GETAUXVAL
 #define ABSL_INTERNAL_USE_GETAUXVAL
+#elif defined(__APPLE__) && defined(ABSL_ARCH_PPC)
+#define ABSL_INTERNAL_USE_PPC_CPUINFO
```

Comment:
I'm having trouble figuring out what this is used for. I see it guards the headers, but I can't tell what code path gets taken or why the headers are needed.

Reply:
I guess because just below it makes an assumption for PPC which is completely wrong on MacOS. It neither uses auxval nor supports ISA v2.07.

Reply:
Sorry, I don't understand this. What does "creates a case to unbreak the build" mean? What breaks without this or if the mach headers are removed?

Reply:
@derekmauro I recall that `randen` created quite a bit of a trouble to fix. I do not remember exact error at the moment.

Reply:
If this is the only stopper, I can rerun the build and either drop this, if it is not required, or give exact justification for having it there.

Reply:
Or I can write the code which extracts PPC cpu type – OpenBLAS has something of this sort.